src/components/Counter.vue

```diff
@@ -3,15 +3,38 @@
     <h1>Counter Component: {{ value }}</h1>
     <div>
       <input type="number" :value="value" @input="$emit('input', $event.target.value)" />
-      <button @click="$emit('increment')">Increment</button>
-      <button @click="$emit('decrement')">Decrement</button>
+      <button @click="increment(value)">Increment</button>
+      <button @click="decrement(value)">Decrement</button>
     </div>
   </div>
 </template>
 
 <script setup>
-const props = defineProps(['value'])
-defineEmits(['input', 'increment', 'decrement'])
+const props = defineProps({
+  value: {
+    type: Number,
+    required: true,
+    default: 0
```

Comment:
@DaudIbrahim You should either specify required or default but not both at the same time in prop definition. Required means that prop has to be provided to the component so it will never be initialized with default value and thus default option becomes redundant.